components/FavoritesButton/index.jsx

```diff
@@ -120,6 +123,9 @@ class FavoritesButton extends Component {
     } else {
       setTimeout(() => {
         this.props.removeFavorites(this.props.productId);
+        if (this.props.useUndo) {
+          this.props.showToast(this.props.productId);
```

Comment:
Instead of trying to control the toast by using the "useUndo" prop, i would suggest to just provide a "onRemove" prop in \<FavoritesButton \/>.
In FavoritesList ->Item we would create a connector that holds the showToast(id) and addFavorites(id, true) actions.
So the FavoriteList\<Item \/> would pass \<FavoritesButton onRemove={this.props.showToast(id)} \/>

This approach is cleaner because the "useUndo" and "showToast" props are only needed in case the button is used in the favorites list :) also the useUndo is completely obsolete then.

Reply:
You're right. Moved to subscription.